app/services/checkout/form_data_adapter.rb

```diff
@@ -34,11 +34,16 @@ def move_payment_source_to_payment_attributes!
     end
 
     def fill_in_card_type
-      payment = params[:order][:payments_attributes]&.first&.dig(:source_attributes)
+      return unless payment_source_attributes
 
-      return if payment&.dig(:number).blank?
+      return if payment_source_attributes.dig(:number).blank?
 
-      payment[:cc_type] ||= card_brand(payment[:number])
+      payment_source_attributes[:cc_type] ||= card_brand(payment_source_attributes[:number])
+    end
+
+    def payment_source_attributes
+      @payment_source_attributes ||=
```

Comment:
I tried to avoid ivars here because they are visible throughout the whole class and it's caching the result for longer than needed. It works in this example but you need to double-check potential other uses when refactoring.

I'm happy to leave it as is but I personally try to reduce visibility as much as possible to avoid conflicts and ease refactoring.

Reply:
Thanks for the feedback!
I see, reducing visibility is important. I thought this improves readability and can also improve performance if we get it re-used in move_payment_source_to_payment_attributes! above.